src/connectivity/mod.rs

```diff
@@ -114,6 +114,79 @@ pub fn strongly_connected_components(graph: &digraph::PyDiGraph) -> Vec<Vec<usiz
         .collect()
 }
 
+fn condensation_inner<N, E, Ty, Ix>(
```

Comment:
This function is called only once in the code, so I don't think you need to make it generic. It is fine to accept only one kind of type for each argument.

n.b: the feedback would be different if this wasn't a private function